api/v1alpha1/progressivesync_types.go

```diff
@@ -17,90 +17,132 @@ limitations under the License.
 package v1alpha1
 
 import (
-	corev1 "k8s.io/api/core/v1"
+	"github.com/Skyscanner/applicationset-progressive-sync/internal/consts"
+	"github.com/fluxcd/pkg/apis/meta"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
-const ProgressiveSyncFinalizer = "finalizers.argoproj.skyscanner.net"
+const (
+	ProgressiveSyncFinalizer = "finalizers.argoproj.skyscanner.net"
+	AppSetKind               = "ApplicationSet"
+)
 
 // ProgressiveSyncSpec defines the desired state of ProgressiveSync
 type ProgressiveSyncSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	// SourceRef defines the resource, example an ApplicationSet, which owns ArgoCD Applications
+	// AppSetRef point to the ApplicationSet which owns ArgoCD Applications
 	//+kubebuilder:validation:Required
-	SourceRef corev1.TypedLocalObjectReference `json:"sourceRef"`
+	AppSetRef meta.LocalObjectReference `json:"appSetRef"`
+
 	// Stages defines a list of Progressive Rollout stages
 	//+kubebuilder:validation:Optional
-	Stages []ProgressiveSyncStage `json:"stages,omitempty"`
+	Stages []Stage `json:"stages,omitempty"`
 }
 
-// ProgressiveSyncStage defines a rollout stage
-type ProgressiveSyncStage struct {
+// Stage defines a rollout stage
+type Stage struct {
 	// Name is a human friendly name for the stage
 	//+kubebuilder:validation:Required
 	Name string `json:"name"`
-	// MaxParallel is how many selected targets to update in parallel
+
+	// MaxParallel is how many targets to sync in parallel
+	//+kubebuilder:validation:Required
 	//+kubebuilder:validation:Minimum:1
-	MaxParallel intstr.IntOrString `json:"maxParallel"`
-	// MaxTargets is the maximum number of selected targets to update
+	MaxParallel int64 `json:"maxParallel"`
```

Comment:
Because I thought the official guidance was to use int64 but it's that or int32 (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types)

I can convert it to int32

Reply:
Discussed offline and decided to keep `int64`